addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6176,13 +6176,21 @@ registry.ImageTools = ImageHandlerOption.extend({
                     img.dataset.originalMimetype = img.dataset.mimetype;
                     img.dataset.mimetype = 'image/svg+xml';
                 }
+                // When the user selects a shape, we remove the data attributes
+                // that are not compatible with this shape.
+                if (saveData && (this._isDeviceShape() || !this._isTransformableShape())) {
```

Comment:
Everywhere you call `_isTransformableShape`, you are doing that `_isDeviceShape` check alongside... should not you put the call to `_isDeviceShape` inside `_isTransformableShape` ?